bundles/org.eclipse.passage.lic.e4.ui/src/org/eclipse/passage/lic/internal/e4/ui/addons/E4LicensingAddon.java

```diff
@@ -46,10 +43,10 @@ public void applicationStarted(//
 			@SuppressWarnings("unused") //
 			@UIEventTopic(UIEvents.UILifeCycle.APP_STARTUP_COMPLETE) //
 			Event event) {
-		ServiceInvocationResult<ExaminationCertificate> result = new EquinoxPassage()
-				.acquireLicense(new LicensedApplicationFromContext(application).get().identifier());
-		if (new ShouldBeExposed().test(result)) {
-			new WorkbenchNotification(() -> context.get(Shell.class)).expose(result);
+		boolean allowed = new EquinoxPassageUI(() -> context.get(Shell.class))
+				.canUse(new LicensedApplicationFromContext(application).get().identifier());
+		if (!allowed) {
+			new WorkbenchShutdown().run();
 		}
 		// FIXME: find a way to call releaseLicense
```

Comment:
`UIEvents.UILifeCycle.APP_SHUTDOWN_STARTED`

Reply:
Will listen it in the subsequent PR.